packages/graphback-datasync/tests/DataSyncMongoDBTests.ts

```diff
@@ -0,0 +1,108 @@
+  }
+  
+  export async function createTestingContext(schemaStr: string, config?: { seedData: { [collection: string]: any[] } }): Promise<Context> {
+      // Setup graphback
+      let schema = buildSchema(schemaStr);
+    
+      const server = new MongoMemoryServer();
+      const client = new MongoClient(await server.getConnectionString(), { useUnifiedTopology: true });
+      await client.connect();
+      const db = client.db('test');
+  
+      const defautConfig = {
+        "create": true,
+        "update": true,
+        "findOne": true,
+        "find": true,
+        "delete": true,
+        "subCreate": true,
+        "subUpdate": true,
+        "subDelete": true
+      }
+    
+      const schemaGenerator = new SchemaCRUDPlugin({ outputPath: './tmp', format: 'graphql'})
+      const DataSyncGenerator = new DataSyncPlugin({ outputPath: './tmp' })
+      const metadata = new GraphbackCoreMetadata({
+        crudMethods: defautConfig
+      }, schema)
+      metadata.setSchema(schemaGenerator.transformSchema(metadata))
+      metadata.setSchema(DataSyncGenerator.transformSchema(metadata));
+      schema = metadata.getSchema();
+    
+      const providers: { [name: string]: DataSyncMongoDBDataProvider } = {}
+      const models = filterModelTypes(schema)
+      for (const modelType of models) {
+        providers[modelType.name] = new DataSyncMongoDBDataProvider(modelType, db);
+      }
+    
+      // if seed data is supplied, insert it into collections
+      if (config?.seedData) {
+        const collectionNames = Object.keys(config.seedData);
+        for (const collectionName of collectionNames) {
+          for (const element of config.seedData[collectionName]) {
+            await providers[collectionName].create(element);
+          }
+        };
+      }
+    
+      return { server, providers }
+    }
+
+describe('Soft deletion test', () => {
+  let context: Context;
+  afterEach(() => context.server.stop());
+
+  it('sets deleted to false on creation', async () => {
+    context = await createTestingContext(`
+    """
+    @model
+    @versioned
+    @delta
+    """
+    type Post {
+      id: ID!
+      text: String!
+    }
+    `);
+
+    const { Post }  = context.providers;
+
+    const first = await Post.create({ text: 'TestPost' });
+    expect(first._deleted).toEqual(false)
+  })
+
+  it('sets deleted to true on deletion', async () => {
+    context = await createTestingContext(`
+    """
+    @model
+    @versioned
+    @delta
+    """
+    type Post {
+      id: ID!
+      text: String!
+    }
+    `);
+
+    const { Post }  = context.providers;
+    const startTime = 1590679886048;
+    const deleteTime = 1590679886048;
+    advanceTo(startTime);
+    const { id } = await Post.create({ text: 'TestPost' });
+    advanceTo(deleteTime);
+    const deletedPost = await Post.delete({ id });
+    expect(deletedPost._deleted).toEqual(true);
+    expect(deletedPost.updatedAt).toEqual(deleteTime);
```

Comment:
I am wondering whether we should have a `deletedAt` field too?

Reply:
I think this has been brought up up before, with conclusion that updatedAt can store this timestamp, @wtrocki what do you think?

Reply:
deletedAt = deleted=true and updatedAt..

Reply:
> I think this has been brought up up before, with conclusion that updatedAt can store this timestamp, @wtrocki what do you think?

Ah sorry, I might have missed the conversation.

Reply:
At this point there is no way to `update` a soft deleted object that I know of,  There was a bug where it could be soft deleted multiple times but I have fixed that now :) Feel free to point out if you see any bugs 🐞

Reply:
Hm.. deletion of deleted item should throw error. edits are imposible

Reply:
Yeap. Edits and I think deletion was possible from the previous patch (before latest fixes) hence my comment.